src/components/KPagination/KPagination.vue

```diff
@@ -207,31 +224,73 @@ const pageSizeOptions = props.pageSizes.map((size, i) => ({
   selected: size === currentPageSize.value,
 }))
 const pageSizeText = computed((): string => `${currentPageSize.value} ${currentPageSize.value === 1 ? 'item per page' : ' items per page'}`)
+/**
+ * KPagination will try to display specified number of neighbors
+ * However, if it will detect overflow, it will try to reduce the number of neighbors to a minimum of 1
+ */
+const fittingNeighbors = ref<number>(props.neighbors)
+/**
+ * By default KPagination tries to display 3 items sequentially
+ * However, if it will detect overflow, it will try to reduce the number of items to a minimum of 1
+ */
+const sequentialItemsVisible = ref<number>(3)
+
+const hasOverflow = async (): Promise<boolean> => {
+  await nextTick() // wait for the DOM to update
+
+  if (!kPaginationElement.value) {
+    return false
+  }
+
+  return kPaginationElement.value.scrollWidth > kPaginationElement.value.clientWidth
+}
+
+/**
+ * Check if the pagination has overflow and reduce the number of neighbors or sequential items visible if possible
+ */
+const fixOverflow = async (): Promise<void> => {
+  const overflowDetected = await hasOverflow()
+
+  if (overflowDetected && (fittingNeighbors.value > 1 || sequentialItemsVisible.value > 1)) {
+    if (fittingNeighbors.value > 1) {
+      fittingNeighbors.value--
+    } else if (sequentialItemsVisible.value > 1) {
+      sequentialItemsVisible.value--
+    }
+
+    pagesVisible.value = getVisiblePages(currPage.value, pageCount.value, firstDetached.value, lastDetached.value)
```

Comment:
suggestion: please add a few comments on the lines in this block explaining the logic

Reply:
Done